dealiiadapter.cc

```diff
@@ -880,83 +1019,93 @@ void ElasticProblem<dim>::initialize_precice()
     interface_nodes_positions.resize(dim * n_interface_nodes);
     interface_nodes_ids.resize(n_interface_nodes);
 
-
-    // get the coordinates of the interface nodes from deal.ii
-    std::map<types::global_dof_index, Point<dim>> support_points;
-    DoFTools::map_dofs_to_support_points(MappingQ1<dim>(), dof_handler, support_points);
-    // support_points contains now the coordinates of all dofs
-    // in the next step, the relevant coordinates are extracted using the extracted coupling_dofs
-    int node_position_iterator=0;
-    for (auto element=coupling_dofs.begin(); element!=coupling_dofs.end(); ++element)
-    {
-        for(int jj=0; jj<dim; ++jj)
-            interface_nodes_positions[node_position_iterator * dim + jj] = support_points[*element][jj];
-
-        ++node_position_iterator;
-    }
-
-    // pass node coordinates to precice
-    precice.setMeshVertices(node_mesh_id, n_interface_nodes, interface_nodes_positions.data(), interface_nodes_ids.data());
-
     // same for the face based mesh
     // number of coupling faces already obtained in the make_grid function
     precice_forces.resize(dim * n_interface_faces);
+
+    // set a constant value for each direction in case of an uncoupled simulation
+    if (parameters.enable_precice == false)
+        for (uint it = 0; it  < precice_forces.size()/dim; it++)
+        {
+            precice_forces[it*dim]= 0;
+            precice_forces[it*dim+1]= 0;
+            if(dim == 3)
+                precice_forces[it*dim+2]= 0;
+        }
+
```

Comment:
Why do we need that? Probably, a better design is possible. If preCICE is not used, `precice_forces` should not be allocated at all.

Reply:
See above